calling-apis/get-requests-with-axios.md

```diff
@@ -0,0 +1,453 @@
+
+```js
+const axios = require("axios");
+
+axios
+  .get("https://us1.locationiq.com/v1/search.php", {
+    params: {
+      key: "... jamel's api key ...",
+      q: "Seattle, Washington, USA",
+      format: "json",
+    },
+  })
+  .then((response) => {
+    console.log("success!", response.data);
+  })
+  .catch((error) => {
+    console.log("error!", error.response.data);
+  });
+```
+
+When he runs it, he indeed gets back the successful response he hoped for!
+
+## Observing Asynchronicity
+
+This API call is made _asynchronously_. This asynchronous nature will affect Jamel approaches his project. But first, how can he observe the asynchronous property of this code?
+
+Let's run this code and observe:
+
+```js
+const axios = require("axios");
+
+axios.get("https://dog.ceo/api/breeds/list/all")
+  .then(() => {
+    console.log('success!');
+  })
+  .catch(() => {
+    console.log('error!');
+  });
+
+console.log("I broke free on a Saturday morning");
+console.log("I put the pedal to the floor");
+console.log("Headed north on Mills Avenue");
+console.log("And listened to the engine roar");
+```
+
+When we run this code, we should see this output:
+
+```
+I broke free on a Saturday morning
+I put the pedal to the floor
+Headed north on Mills Avenue
+And listened to the engine roar
+success!
+```
+
+Our console prints out the lyrics at the bottom first, **before** our API call prints "success!" But our song lyrics were written _below_ the lines where we made our API call.
+
+Our JavaScript still gets executed from top to bottom. However, when we make our GET request, our program **does not wait for the HTTP response to come back**. Instead, after sending the `GET` request, the program **proceeds** and continues to execute the next lines of code.
+
+### Working With Asynchronous Code
+
+Because this is asynchronous, we sometimes need to tailor our code to anticipate that our response may come back _after_ the rest of our program executes.
+
+Observe this bug:
+
+```js
+const axios = require("axios");
+
+const dogGreeting = "bark";
+let dogImageURL;
+
+axios
+  .get("https://dog.ceo/api/breeds/image/random")
+  .then((response) => {
+    dogImageURL = response.data.message;
+  })
+  .catch((error) => {
+    dogImageURL = "There was an error";
+  });
+
+console.log(`My favorite dog greeting is ${dogGreeting}`);
+console.log(`A random dog image URL is ${dogImageURL}`);
+```
+
+Reading through this code from top-to-bottom, without understanding asynchronosity, would lead us to believe that:
+
+1. We declare `let dogImageURL` at the top
+1. **Then,** we make an API call, and then assign the value of the API's response data to `dogImageURL`
+1. **After that**, we print to the terminal with `console.log` the value of `dogImageURL`
+
+However, because the API call happens asynchronously, our program doesn't wait for the API response to come back before proceeding with the program. The result is that our `dogImageURL` comes back as `undefined`.
```

Comment:
asynchronosity typo
reorder the sentence describing the print statement (keep the direct object closer to the verb)
at the point when the print occurs, the result hasn't yet "come back", so restate to indicate that it hasn't been changed yet

```suggestion
Reading through this code from top-to-bottom, without understanding asynchronicity, would lead us to believe that:

1. We declare `let dogImageURL` at the top
1. **Then,** we make an API call, and assign the value of the API's response data to `dogImageURL`
1. **After that**, we print the value of `dogImageURL` to the terminal using `console.log()`

However, because the API call happens asynchronously, our program doesn't wait for the API response to come back before proceeding with the program. The result is that when we try to output our `dogImageURL` it is still `undefined`!
```